### PR TITLE
Add capacity mgmt module for Az

### PR DESCRIPTION
**Pull request by GeorgeH-MS**
Add capacity management module to Az branch

### CODE REVIEW

**Comment by GeorgeH-MS** [comment]
@NeilBird @XiangyuKuangMSFT could you help to review this pull request?

**Comment by NeilBird** [comment]
Looks good, adding the Storage Performance Dashboard Creation script to the Az branch.

File: CapacityManagement/DashboardGenerator/Create-AzSStorageDashboard.ps1 [hunk 1]
```diff
@@ -0,0 +1,544 @@
...
+        [datetime]$endTime,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet('Automatic', 'PT1M', 'PT1H', 'P1D', 'PT5M', 'PT15M', 'PT30M', 'PT6H', 'PT12H')]
+        [string]$timeGrain = 'Automatic',
+        [Parameter(Mandatory = $false)]
+        [string]$outputLocation = '.'
+    )
+
+    # check if AzureRM Module is present
+    $AzureRMModule = Get-command -Name Get-AzureRmContext -ErrorAction SilentlyContinue
```
Comment (bganapa) [inline CapacityManagement/DashboardGenerator/Create-AzSStorageDashboard.ps1:83]: this should change to Get-AzContext , there should not be any reference to -AzureRM* in this script and -AzureRM should be replaced with -Az

File: CapacityManagement/DashboardGenerator/Create-AzSStorageDashboard.ps1 [hunk 2]
```diff
@@ -0,0 +1,544 @@
...
+}
+
+function Get-AzureStackResourceId {
+    [CmdletBinding()]
+    param (
+    )
+    try {
+        Write-Host "Getting resource Id from AzureRmSubscription."
+        $adminSubscription = $script:context.Subscription
+        $location = Get-AzureRmLocation -DefaultProfile $script:context -ErrorAction Stop -Verbose
```
Comment (bganapa) [inline CapacityManagement/DashboardGenerator/Create-AzSStorageDashboard.ps1:197]: @GeorgeH-MS  pls rreplace -AzureRM with -Az

File: CapacityManagement/DashboardGenerator/Create-AzSStorageDashboard.ps1 [hunk 3]
```diff
@@ -0,0 +1,544 @@
...
+    }
+    catch {
+        Write-Error $_
+        throw "Please login in AzureRm or Az account. If still happens, check your environment settings in psm1 file."
```
Comment (bganapa) [inline CapacityManagement/DashboardGenerator/Create-AzSStorageDashboard.ps1:201]: Please login with Login-AzAcoount

File: CapacityManagement/DashboardGenerator/Create-AzSStorageDashboard.ps1 [hunk 4]
```diff
@@ -0,0 +1,544 @@
...
+                }
+            }
+        } else {
+            throw "Error: This script requires either the 'AzureRM' or 'Az' PowerShell Module to be installed."
+        }
+    }
+
+    # If user do not input DefaultProfile
+    if ($null -eq $DefaultProfile) {
+        $script:context = Get-AzureRmContext
```
Comment (bganapa) [inline CapacityManagement/DashboardGenerator/Create-AzSStorageDashboard.ps1:118]: Get-AzContext

**Comment by GeorgeH-MS** [comment]
Updated Create-AzSStorageDashboard.ps1, removed all AzureRM related content.

**Review by bganapa — APPROVED** [review]

**Pull request merged**